enterprise/internal/campaigns/resolvers/patch_sets_test.go

```diff
@@ -0,0 +1,341 @@
+	testHeadRevision := "b69072d5f687b31b9f6ae3ceafdc24c259c4b9ec"
+
+	repo := &repos.Repo{ID: api.RepoID(1), Name: "github.com/sourcegraph/sourcegraph"}
+
+	backend.Mocks.Repos.ResolveRev = func(_ context.Context, _ *types.Repo, rev string) (api.CommitID, error) {
+		if rev != testBaseRevision && rev != testHeadRevision {
+			t.Fatalf("ResolveRev received wrong rev: %q", rev)
+		}
+		return api.CommitID(rev), nil
+	}
+	defer func() { backend.Mocks.Repos.ResolveRev = nil }()
+
+	backend.Mocks.Repos.GetCommit = func(_ context.Context, _ *types.Repo, id api.CommitID) (*git.Commit, error) {
+		if string(id) != testBaseRevision && string(id) != testHeadRevision {
+			t.Fatalf("GetCommit received wrong ID: %s", id)
+		}
+		return &git.Commit{ID: id}, nil
+	}
+	defer func() { backend.Mocks.Repos.GetCommit = nil }()
+
+	patch := &patchResolver{
+		store: ee.NewStoreWithClock(dbconn.Global, clock),
+		patch: &campaigns.Patch{
+			RepoID:  repo.ID,
+			Rev:     api.CommitID(testHeadRevision),
+			BaseRef: testBaseRevision,
+			Diff:    testDiff,
+		},
+		preloadedRepo: repo,
+	}
+
+	t.Run("Pagination", func(t *testing.T) {
+		endCursors := []string{"1", "2"}
+		totalCount := int32(testDiffFiles)
+
+		tests := []struct {
+			first int32
+			after string
+
+			wantNodeCount int
+
+			wantTotalCount *int32
+
+			wantHasNextPage bool
+			wantEndCursor   *string
+		}{
+			{
+				first:           1,
+				after:           "",
+				wantNodeCount:   1,
+				wantHasNextPage: true,
+				wantEndCursor:   &endCursors[0],
+				wantTotalCount:  nil,
+			},
+			{
+				first:           1,
+				after:           endCursors[0],
+				wantNodeCount:   1,
+				wantHasNextPage: true,
+				wantEndCursor:   &endCursors[1],
+				wantTotalCount:  nil,
+			},
+			{
+				first:           1,
+				after:           endCursors[1],
+				wantNodeCount:   1,
+				wantHasNextPage: false,
+				wantEndCursor:   nil,
+				wantTotalCount:  &totalCount,
+			},
+			{
+				first:           testDiffFiles + 1,
+				after:           "",
+				wantNodeCount:   testDiffFiles,
+				wantHasNextPage: false,
+				wantEndCursor:   nil,
+				wantTotalCount:  &totalCount,
+			},
+		}
+
+		for _, tc := range tests {
+			args := &graphqlbackend.FileDiffsConnectionArgs{First: &tc.first}
+			if tc.after != "" {
+				args.After = &tc.after
+			}
+
+			conn, err := patch.FileDiffs(ctx, args)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			nodes, err := conn.Nodes(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if len(nodes) != tc.wantNodeCount {
+				t.Fatalf("wrong length of nodes. want=%d, have=%d", tc.wantNodeCount, len(nodes))
+			}
+
+			pageInfo, err := conn.PageInfo(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if pageInfo.HasNextPage() != tc.wantHasNextPage {
+				t.Fatalf("pageInfo HasNextPage wrong. want=%t, have=%t", tc.wantHasNextPage, pageInfo.HasNextPage())
+			}
+
+			if diff := cmp.Diff(pageInfo.EndCursor(), tc.wantEndCursor); diff != "" {
+				t.Fatal(diff)
+			}
```

Comment:
IIRC our current convention is to put "want" as first argument:
```suggestion
			if diff := cmp.Diff(tc.wantEndCursor, pageInfo.EndCursor()); diff != "" {
				t.Fatalf("(-want +got):\n%s", diff)
			}
```

Same for other occurrences.